be/test/runtime/kafka_consumer_pipe_test.cpp

```diff
@@ -36,6 +36,12 @@
 
 #include <gtest/gtest.h>
 
+#include "testutil//assert.h"
+
+// let gcov to calculate code coverage for kafka_consumer_pipe.h
+#pragma GCC push_options
+#pragma GCC optimize("no-inline")
```

Comment:
will this be working for clang? If this is only for code coverage inaccuracy, it should be acceptable to miss the lines, just add a few comments there.

Reply:
UTs are added to improve the code quality, code coverage is just a metrics to measure the quality, not the target goal, lol.

Reply:
just try to meet the code merge requirement. This solution refers to [observer_test.cpp](https://github.com/StarRocks/starrocks/blob/main/be/test/exec/pipeline/schedule/observer_test.cpp#L40), but seems moving `copy_from` implementation to byte_buffer.cpp solves the problem, so just remove  these GCC options